07_errors/mohitnag/mapstore.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"strconv"
+)
+
+// CreatePuppy creates a Puppy in mapstore
+func (m MapStore) CreatePuppy(p Puppy) error {
+	if _, ok := m[p.ID]; !ok {
+		val, _ := strconv.Atoi(p.Value)
+		if val < 0 {
+			return ErrorF(InvalidValue, "puppy with value less than 0 not allowed")
+		}
+		m[p.ID] = p
+		return nil
+	}
+	return ErrorF(Duplicate, "puppy with Id %d already exists", p.ID)
```

Comment:
Imo invert logic: if ok... Then error.
It's more consistent with the ifs usually containing the error case and "bail early on error"

Reply:
Completely agree.